src/lib/mina_lib/mina_lib.ml

```diff
@@ -1632,8 +1642,15 @@ let fetch_completed_snarks (module Context : CONTEXT) snark_pool network
       Deferred.unit )
     limited_peers
 
+let initialization_error_to_exn (`Initialization_error e) =
+  Error.to_exn @@ Error.tag ~tag:"proof cache initialization error" e
+
+let initialize_proof_cache_db (config : Config.t) =
+  Proof_cache_tag.create_db ~logger:config.logger
+    (config.conf_dir ^/ "proof_cache")
+  >>| function Error e -> raise (initialization_error_to_exn e) | Ok db -> db
+
```

Comment:
Nit: The `Error.to_exn` followed immediately by `raise` is a little odd, vs a direct `Error.raise`. Stylistically, I prefer this, but it's otherwise equivalent
```suggestion
let initialize_proof_cache_db (config : Config.t) =
  match%map
    Proof_cache_tag.create_db ~logger:config.logger
      (config.conf_dir ^/ "proof_cache")
  with
  | Error e ->
      Error.raise @@ Error.tag ~tag:"proof cache initialization error" e
  | Ok db ->
      db
```

Reply:
Didn't notice the existence of `Error.raise`, works for me